_saas-integrations/bing-ads/v2/bing-ads-v2.md

```diff
@@ -92,31 +98,46 @@ setup-steps:
 # -------------------------- #
 
 replication-sections:
-  - content: |
-      {% include integrations/saas/ads-append-only-replication.html type="table-types" %}
-
-  - title: "Report Tables: Data Extraction & Conversion Windows"
-    anchor: "data-extraction-conversion-window"
+  - title: "Report retention windows and historical data"
+    anchor: "retention-window-historical-data"
     content: |
-      {% include integrations/saas/ads-append-only-replication.html type="report-tables" %}
+      {{ integration.display_name }} retains reporting data for specified periods of time. Depending on the type of report, the retention period can vary. For the majority of reports, however, Bing Ads will retain data for **36 months** from the current date.
 
-  - title: "Report Tables: Data Loading & Append-Only Replication"
-    anchor: "data-loading-append-only"
-    content: |
-      {% include integrations/saas/ads-append-only-replication.html type="data-loading" %}
+      This retention window can impact the historical data Stitch can replicate from {{ integration.display_name }}. If you select a **Start Date** further in the past than the retention window allows, an error similar to the following will surface in the integration's [Extraction Logs]({{ link.replication.extraction-logs | prepend: site.baseurl }}) during the extraction phase:
 
-  - title: "Report Tables: Query for the Latest Data"
-    anchor: "query-for-the-latest-data"
+      ```
+      tap - Message = "The specified report time contains an invalid custom date range end. Please submit a report request with valid start and end dates for the custom date range."
+      ```
```

Comment:
Ah! I guess I didn't write this down anywhere, but the tap shouldn't throw an error in this case anymore. If it runs into it, it will log [this message](https://github.com/singer-io/tap-bing-ads/blob/master/tap_bing_ads/__init__.py#L720-L721) and move the date range up a month until it succeeds.

Reply:
Updated!